src/components/EResourceSections/AcquisitionOptions/AcquisitionOptions.js

```diff
@@ -6,55 +6,74 @@ import { Accordion, Badge, MultiColumnList, NoValue, Spinner } from '@folio/stri
 import {
   EResourceType,
   isPackage,
-  TitleOnPlatformLink
+  TitleOnPlatformLink,
+  usePrevNextPagination
 } from '@folio/stripes-erm-components';
 import AddToBasketButton from '../../AddToBasketButton';
 import Coverage from '../../Coverage';
 import EResourceKB from '../../EResourceKB';
 
-import { isExternal } from '../../utilities';
-import { resourceClasses } from '../../../constants';
-
-class AcquisitionOptions extends React.Component {
-  static propTypes = {
-    data: PropTypes.shape({
-      areEntitlementOptionsLoading: PropTypes.bool,
-      entitlementOptions: PropTypes.arrayOf(PropTypes.object),
-      entitlementOptionsCount: PropTypes.number,
-      eresource: PropTypes.shape({
-        name: PropTypes.string,
-      }),
+import { isExternal, parseMclPageSize } from '../../utilities';
+import { resourceClasses, ENTITLEMENT_OPTIONS_PAGINATION_ID } from '../../../constants';
+import { useAgreementsSettings } from '../../../hooks';
+
+const propTypes = {
+  data: PropTypes.shape({
+    areEntitlementOptionsLoading: PropTypes.bool,
+    entitlementOptions: PropTypes.arrayOf(PropTypes.object),
+    entitlementOptionsCount: PropTypes.number,
+    eresource: PropTypes.shape({
+      name: PropTypes.string,
     }),
-    handlers: PropTypes.shape({
-      onEResourceClick: PropTypes.func,
-      onNeedMoreEntitlementOptions: PropTypes.func,
-    }),
-    id: PropTypes.string,
-  };
-
-  getName = (eresource = {}) => {
+  }),
+  handlers: PropTypes.shape({
+    onEResourceClick: PropTypes.func,
+    onNeedMoreEntitlementOptions: PropTypes.func,
+  }),
+  id: PropTypes.string,
+  isLoading: PropTypes.bool.isRequired,
+};
+
+const AcquisitionOptions = ({
+  data: { areEntitlementOptionsLoading, entitlementOptions, entitlementOptionsCount },
+  handlers: { onEResourceClick, onNeedMoreEntitlementOptions },
+  id,
+  isLoading
+}) => {
+  const { settings } = useAgreementsSettings();
+  const entitlementOptionsPageSize = parseMclPageSize(settings, 'entitlements');
+
+  const {
+    paginationMCLProps,
+  } = usePrevNextPagination({
+    count: entitlementOptionsCount,
+    pageSize: entitlementOptionsPageSize,
+    id: ENTITLEMENT_OPTIONS_PAGINATION_ID,
+    syncToLocation: false
+  });
+
+  const getName = (eresource = {}) => {
     if (isExternal(eresource)) {
       return eresource.reference_object?.label;
     }
 
     return eresource.name;
-  }
+  };
 
-  onRowClick = (_, row) => {
-    const { id, class: clazz } = row;
-    const { handlers: { onEResourceClick } } = this.props;
+  const onRowClick = (_, row) => {
+    const { class: clazz } = row;
 
     // Redirect to Package view if package, title view otherwise
     onEResourceClick(id, clazz === resourceClasses?.PACKAGE ? 'PKG' : 'TITLE');
```

Comment:
This id is _not_ the same as the id pased by row... really subtle bug this one :/